luda-editor/luda-editor-client/src/app/editor/job/move_camera_clip.rs

```diff
@@ -11,139 +11,142 @@ pub struct MoveCameraClipJob {
                 }
-            } else {
-                if clip_center_at < moved_end_at {
-                    next_moving_clip_index = index;
-                }
-            }
-        }
-
-        if moving_clip_index != next_moving_clip_index {
-            let min = moving_clip_index.min(next_moving_clip_index);
-            let max = moving_clip_index.max(next_moving_clip_index);
+            })
+            .flatten()
+            .collect()
+    }
 
-            let slice_to_rotate = &mut clips[min..max + 1];
-            if moving_clip_index < next_moving_clip_index {
-                slice_to_rotate.rotate_left(1);
-            } else {
-                slice_to_rotate.rotate_right(1);
-            }
+    fn mock_sequence(camera_clip_ids: &[&str]) -> Sequence {
+        let mut clips = Vec::new();
+        camera_clip_ids.iter().enumerate().for_each(|(index, id)| {
+            let start_at = Time::from_ms(index as f32);
+            let end_at = Time::from_ms((index + 1) as f32);
+            clips.push(mock_camera_clip(id, start_at, end_at));
+        });
+        Sequence {
+            tracks: vec![Arc::new(Track::Camera(CameraTrack {
+                id: "track-1".to_string(),
+                clips: clips.into(),
+            }))]
+            .into(),
         }
+    }
 
-        push_front_camera_clips(&mut clips);
-
-        if is_preview {
-            clips
-                .iter_mut()
-                .find(|clip| clip.id.eq(moving_clip_id))
-                .map(|moving_clip| {
-                    let new_clip = Arc::new(CameraClip {
-                        id: moving_clip.id.clone(),
-                        start_at: moved_start_at,
-                        end_at: moved_end_at,
-                        camera_angle: moving_clip.camera_angle.clone(),
-                    });
-                    let _ = std::mem::replace(moving_clip, new_clip);
-                });
-        }
-        camera_track.clips = clips.into();
+    fn mock_camera_clip(clip_id: &str, start_at: Time, end_at: Time) -> Arc<CameraClip> {
+        Arc::new(CameraClip {
+            id: clip_id.to_string(),
+            start_at,
+            end_at,
+            camera_angle: CameraAngle {
+                character_pose_emotion: CharacterPoseEmotion(
+                    "c".to_string(),
+                    "p".to_string(),
+                    "e".to_string(),
+                ),
+                source_01_circumscribed: Circumscribed {
+                    center: Xy { x: 0.0, y: 0.0 },
+                    radius: 1.0,
+                },
+                crop_screen_01_rect: LtrbRect {
+                    left: 0.0,
+                    top: 0.0,
+                    right: 1.0,
+                    bottom: 1.0,
+                },
+            },
+        })
     }
 }
 
-fn replace_track(sequence: &mut Sequence, track: CameraTrack) {
-    let track_index = sequence
+pub(crate) fn get_camera_clip_id(sequence: &Sequence) -> String {
```

Comment:
```suggestion
pub(crate) fn get_camera_track_id(sequence: &Sequence) -> String {
```
related to https://github.com/NamseEnt/namseent/pull/96/files#r796675567